### PR TITLE
a11y: Remove "copy" repetition

### PR DESCRIPTION
**Pull request by yoomlam**
Follows PR #16293 (and obsolete PR #16258)

### Description
This PR was started by Yoom to resolve a problem where a screenreader would say "Copy     Copy...." for some CopyTextButtons on the hearing details page for a virtual hearing. Charley adopted this PR and rewrote the comment.

#### Before this PR
- The `aria-label` property on two CopyTextButtons on a hearing details page in the "Virtual Hearing" section is `Copy     Copy VLJ Link` and `Copy    Copy Veteran Link`. Screen readers reading that sound strange.
- The CopyTextButton component builds an `aria-label` like this: `Copy ${label} ${text}` .
- Unfortunately, the `HearingLinks` component uses the word "Copy" in the button text like: `label = Copy VLJ Link`.
- That meant that we'd end up with an arial-label like `Copy    Copy VLJ Link` which was being read by the screenreader.

#### After this PR
- The `aria-label` property on two CopyTextButtons on a hearing details page in the "Virtual Hearing" section would end up being `Copy VLJ Link` and `Copy Veteran Link`. Screen readers reading that sound correct.
- CopyTextButton has an additional prop: `ariaLabel`. This prop is only used by `HearingLinks`.
- The CopyTextButton component builds an `aria-label` like this: `Copy ${label} ${text}` *when we don't pass the new ariaLabel prop*. This preserves existing behavior.
- When `ariaLabel` is set, we use that string for `aria-label`.

### Acceptance Criteria
- [ ] Code compiles correctly
- [ ] screenreader does not say "Copy Copy" on relevant components

### Testing Plan
Make sure screenreader does not say "Copy Copy" on relevant components

### CODE REVIEW

File: client/test/app/hearings/components/__snapshots__/Details.test.js.snap [hunk 1]
```diff
@@ -62248,37 +62248,37 @@ exports[`Details Displays VirtualHearing details when there is a virtual hearing
...
+                                    data-for="tooltip-Guest Link"
                                     data-tip={true}
                                     tabIndex={0}
                                     text="https://care.evn.va.gov/bva-app/?join=1&media=&escalate=1&conference=BVA0000009@care.evn.va.gov&pin=2684353125#&role=guest"
                                   >
                                     <button
-                                      aria-describedby="tooltip-Copy Guest Link"
-                                      aria-label="Copy  Copy Guest Link"
+                                      aria-describedby="tooltip-Guest Link"
+                                      aria-label="Copy  Guest Link"
```
Comment (yoomlam) [inline client/test/app/hearings/components/__snapshots__/Details.test.js.snap:62270]: Repetition removed on `aria-label`.

**Review by jcq — COMMENTED** [review]
Looks like a good improvement! Snapshots seem to confirm it's changing things in the desired way across multiple usages.

**Comment by yoomlam** [comment]
Unfortunately, Rspecs expects "Copy ... Link" buttons:
https://github.com/department-of-veterans-affairs/caseflow/blob/9e10c99ba1f611a9262e04957913e0c81bc3f5f4/spec/feature/hearings/virtual_hearings/hearing_details_spec.rb#L102
https://github.com/department-of-veterans-affairs/caseflow/blob/9e10c99ba1f611a9262e04957913e0c81bc3f5f4/spec/feature/hearings/virtual_hearings/hearing_details_spec.rb#L112

**Review by jcq — APPROVED** [review]
Good improvement — it's always better to allow optional overriding of props like this

**Pull request merged**